File: Java/src/main/java/com/namevalidator/NameValidator.java
package com.namevalidator;
import java.util.regex.Pattern;
public class NameValidator {

	private static final String regex = "^[a-zA-Z0-9]+$";
	
	public boolean validateUserName(String name) {
        if (name.length() < 6 || name.length() > 12) return false;
        //initializes the regex pattern, gives it the username, and returns true or false if they match
        return Pattern.compile(regex).matcher(name).matches();
    }
}

File: csharp/NameValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace NameValidator_UT_Kata
{
    public class NameValidator
    {
        private const string regex = "^[a-zA-Z0-9]+$";

        public bool ValidateUserName(string name)
        {
            if (name.Length < 6 || name.Length > 12) return false;
            // initializes the regex pattern, gives it the username, and returns true or false if they match
            return Regex.IsMatch(name, regex);
        }
    }
}
